src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/AvoidUninstantiatedInternalClasses.cs

```diff
@@ -314,7 +314,7 @@ private static bool ContainsEntryPoint(INamedTypeSymbol type, Compilation compil
                 return false;
             }
 
-            var taskSymbol = compilation.GetTypeByMetadataName("System.Threading.Tasks.Task");
+            var taskSymbol = WellKnownTypes.Task(compilation);
             var genericTaskSymbol = compilation.GetTypeByMetadataName("System.Threading.Tasks.Task`1");
```

Comment:
We should add a new well known type for `GenericTask`. I will do so in a separate follow-up change.